File: Scala/src/main/scala/org/revian/MyArray.scala
package org.revian

class MyArray(capacity: Int) {
  private var arr = new Array[Int](capacity)
  private var arraySize = 0

  def add(value: Int): Unit = {
    if (arraySize == arr.length) {
      throw new ArrayIndexOutOfBoundsException("Array is full!")
    }
    arr(arraySize) = value
    arraySize += 1
  }

  def get(index: Int): Int = {
    if (index >= arraySize) throw new IndexOutOfBoundsException("Index out of bounds")
    arr(index)
  }

  def size(): Int = arraySize
}

File: Java/src/main/java/org/revian/MyArray.java
package org.revian;

public class MyArray {
    private int[] array;
    private int size;

    public MyArray(int capacity) {
        array = new int[capacity];
        size = 0;
    }

    public void add(int value) {
        if (size >= array.length) {
            throw new IndexOutOfBoundsException("Array is full");
        }
        array[size++] = value;
    }

    public int get(int index) {
        if (index < 0 || index >= size) {
            throw new IndexOutOfBoundsException("Invalid index");
        }
        return array[index];
    }

    public int size() {
        return size;
    }
}
